src/Arcus.Testing.Tests.Integration/Integration/DataFactory/Fixture/TemporaryDataFactoryDataFlow.cs

```diff
@@ -159,38 +203,48 @@ private async Task AddCsvSourceAsync(AssertCsvOptions options)
+
             await _sourceDataset.UpdateAsync(WaitUntil.Completed, new DataFactoryDatasetData(sourceProperties));
         }
 
         private async Task AddJsonSourceAsync()
         {
-            _logger.LogTrace("Adding JSON source '{SourceName}' to DataFlow '{DataFlowName}' within Azure DataFactory '{DataFactoryName}'", SourceName, Name, DataFactory.Name);
+            _logger.LogTrace("Adding JSON source '{SourceName}' to DataFlow '{DataFlowName}' within Azure DataFactory '{DataFactoryName}'", SourceDataSetName, Name, DataFactory.Name);
 
             var blobStorageLinkedService = new DataFactoryLinkedServiceReference(DataFactoryLinkedServiceReferenceKind.LinkedServiceReference, _linkedServiceName);
 
-            _sourceDataset = _arm.GetDataFactoryDatasetResource(DataFactoryDatasetResource.CreateResourceIdentifier(SubscriptionId, ResourceGroupName, DataFactory.Name, SourceName));
+            _sourceDataset = _arm.GetDataFactoryDatasetResource(DataFactoryDatasetResource.CreateResourceIdentifier(SubscriptionId, ResourceGroupName, DataFactory.Name, SourceDataSetName));
 
             var sourceProperties = new JsonDataset(blobStorageLinkedService)
             {
                 DataLocation = new AzureBlobStorageLocation()
                 {
                     Container = _sourceContainer?.Name ?? throw new InvalidOperationException("Azure blob storage container should be available at this point"),
-                    FolderPath = SourceName
+                    FolderPath = SourceDataSetName
                 }
             };
             await _sourceDataset.UpdateAsync(WaitUntil.Completed, new DataFactoryDatasetData(sourceProperties));
         }
 
-        private async Task AddCsvSinkAsync(AssertCsvOptions options)
+        private async Task AddCsvSinkAsync(AssertCsvOptions options, TempDataFlowOptions dataFlowOptions)
         {
-            _logger.LogTrace("Adding CSV sink '{SinkName}' to DataFlow '{DataFlowName}' within Azure DataFactory '{DataFactoryName}'", _sinkDatasetName, Name, DataFactory.Name);
+            _logger.LogTrace("Adding CSV sink '{SinkName}' to DataFlow '{DataFlowName}' within Azure DataFactory '{DataFactoryName}'", SinkDataSetName, Name, DataFactory.Name);
 
             var blobStorageLinkedService = new DataFactoryLinkedServiceReference(DataFactoryLinkedServiceReferenceKind.LinkedServiceReference, _linkedServiceName);
 
-            _sinkDataset = _arm.GetDataFactoryDatasetResource(DataFactoryDatasetResource.CreateResourceIdentifier(SubscriptionId, ResourceGroupName, DataFactory.Name, _sinkDatasetName));
+            _sinkDataset = _arm.GetDataFactoryDatasetResource(DataFactoryDatasetResource.CreateResourceIdentifier(SubscriptionId, ResourceGroupName, DataFactory.Name, SinkDataSetName));
+
+            foreach (var parameter in dataFlowOptions.Sink.SinkDataSetParameterKeyValues)
+            {
+                SinkDataSetParameterKeyValues.Add(parameter.Key, parameter.Value);
+            }
```

Comment:
There is actually now no need for the `TemporaryDataFactoryDataFlow` fixture to be having that dictionary as a property, One can call here directly the `dataFlowOptions.Sink.ApplyOptions()` and let the options add those parameters instead of having two different sets of dictionaries doing the same thing.

Reply:
Fixed in [#6decbd3](https://github.com/arcus-azure/arcus.testing/pull/228/commits/6decbd3f9512b20dd599201881b5773d01cf09f9)